### ISSUE TITLE
Package motivation

### ISSUE THREAD

**Issue by judocode** [stringsync/vexml#219]
Hi, I stumbled across this as a consumer of Vexflow and I was curious about the difference between this package and https://github.com/opensheetmusicdisplay/opensheetmusicdisplay as it appears on the surface they are solving the same problem?

**Comment by jaredjj3** [comment]
Hey @judocode, thanks for your interest!

You're correct — OSMD and vexml are trying to solve the same problem: render MusicXML in a browser. OSMD is a solid library with a great community and I recommend that you try it if you haven't already. I built stringsync.com with it!

However, I wasn't happy with [what I built on top of OSMD](https://github.com/stringsync/stringsync/tree/c65bc9d10f5392ac69b1700488393340ff0686b2/web/src/lib/MusicDisplay). I sought after better support for interactivity in https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/issues/1056, but this would be a huge undertaking and it didn't gain much traction. Around this time, I started to become more interested in the idea of completely encoding music, which is precisely what MusicXML does.

Eventually I decided to create [stringsync/musicxml](https://github.com/stringsync/musicxml) to make it easier to manipulate MusicXML in a typesafe manner. @rvilarl reached out to me in https://github.com/stringsync/musicxml/issues/1 to understand why I was creating that library. That conversation lead to the beginning of this repo.

@rvilarl and I have been developing this very casually over the past 2 years. We've been finding issues with VexFlow, so it's helped in the development of that. Ultimately, I think OSMD and vexml will have a lot of overlap but will have their own strengths and weaknesses.

If you don't mind sharing, I'd like to understand your needs better. What features would you want from a MusicXML rendering library?

**Comment by judocode** [comment]
Appreciate you taking the time to explain! I think the "why" would help to be included in the Readme so folks understand the motivation and what this is attempting to solve. I'm after solving similar problems which I partially detailed in [this comment on the Vexflow 5 thread](https://github.com/vexflow/vexflow/issues/13#issuecomment-2076822223).

TLDR my ideal is a scenario where:
- A full JSON spec similar to MusicXML is established (aka MusicJSON -- could be a simple 1:1 as much as it can be with XML to JSON)
- A library to convert to and from MusicXML <-> MusicJSON (looks like that's essentially what you are doing with [musicxml](https://github.com/stringsync/musicxml))
- Vexflow having the ability to accept MusicJSON directly, so I can simply give it an object and know it will render everything without needing to dig into additional APIs
- Vexflow is responsive by default (adjusts scale and width based on size of its container if width/height are not supplied)
- Ideally interactivity is built in as well, but I could definitely see that being its own package on top of Vexflow

I've implemented my own versions of all of the above (in a closed-source project), as I'm sure many others have, but I'd like to help bring these either to Vexflow itself or to other open source projects.

Like you, I also believe EasyScore isn't quite the right abstraction, and it feels once it is extended, then it just becomes what the other APIs in vexflow already solve. I feel there is a middle ground where things are Object-driven rather than api-driven and in a full-spec directly translatable to MusicXML.

Also worth mentioning, I too am using React, so taking things a step further, I am ultimately after leveraging all of this functionality to help make building a responsive music notation web app in React more easy.

**Comment by jaredjj3** [comment]
>Appreciate you taking the time to explain! I think the "why" would help to be included in the Readme

Done. https://github.com/stringsync/vexml/pull/220

>A full JSON spec similar to MusicXML is established (aka MusicJSON -- could be a simple 1:1 as much as it can be with XML to JSON)

I believe this is what [MNX](https://w3c.github.io/mnx/docs/) is trying to be. Otherwise, I recommend checking out [stringsync/musicxml](https://github.com/stringsync/musicxml) for a typesafe way to navigate a MusicXML document in TypeScript.

However, if you _really_ just want a 1:1 conversion (maybe you have a storage constraint?), check out this [gist](https://gist.github.com/jaredjj3/4d2caabb28037d66f284eaea16d7154f) I wrote. You could do the backward translation if you want.

>Vexflow is responsive by default (adjusts scale and width based on size of its container if width/height are not supplied)

I agree, it would be amazing if `VexFlow` supported this. However, it doesn't have a concept of musical systems that I know of, which makes this task a large effort. I briefly talk about some of this differences in

**Comment by jaredjj3** [comment]
@judocode, just FYI. I rebuilt the rendering engine to be agnostic of the music encoding. As long as you can conform your encoding to [data.Score](https://github.com/stringsync/vexml/blob/8ff837ad3f4c1f1764f7e922faac255860e981d6/src/data/types.ts#L28), you can use vexml's rendering engine. You can trace [MusicXMLParser](https://github.com/stringsync/vexml/blob/8ff837ad3f4c1f1764f7e922faac255860e981d6/src/parsing/musicxml/musicxmlparser.ts) to see how I did it for MusicXML.
